tests/ha/test_control_pod_restart.py

```diff
@@ -0,0 +1,417 @@
+        self.restore_pod = self.restore_method = self.deployment_name = None
+        self.deployment_backup = None
+        if not os.path.exists(self.test_dir_path):
+            sysutils.make_dirs(self.test_dir_path)
+        self.multipart_obj_path = os.path.join(self.test_dir_path, self.test_file)
+        LOGGER.info("Updating control pod deployment yaml")
+        self.control_pods = self.node_master_list[0].get_pods_node_fqdn(
+            const.CONTROL_POD_NAME_PREFIX)
+        self.control_pod_name = list(self.control_pods.keys())[0]
+        self.control_node = self.control_pods.get(self.control_pod_name)
+        resp = self.ha_obj.update_deployment_yaml(pod_obj=self.node_master_list[0],
+                                                  pod_name=self.control_pod_name,
+                                                  find_key="persistentVolumeClaim",
+                                                  replace_key="emptyDir", replace_val=dict())
+        assert_utils.assert_true(resp[0], resp)
+        self.modified_yaml = resp[1]
+        self.backup_yaml = resp[2]
+        LOGGER.info("Done: Setup operations.")
+
+    def teardown_method(self):
+        """
+        This function will be invoked after each test function in the module.
+        """
+        LOGGER.info("STARTED: Teardown Operations.")
+        if self.s3_clean:
+            LOGGER.info("Cleanup: Cleaning created s3 accounts and buckets.")
+            resp = self.ha_obj.delete_s3_acc_buckets_objects(self.s3_clean)
+            assert_utils.assert_true(resp[0], resp[1])
+        if self.restore_pod:
+            LOGGER.info("Restoring control pod to its original state using yaml file %s",
+                        self.original_backup)
+            control_pod_name = self.node_master_list[0].get_all_pods(
+                const.CONTROL_POD_NAME_PREFIX)[0]
+            pod_yaml = {control_pod_name: self.original_backup}
+            resp = self.ha_obj.failover_pod(pod_obj=self.node_master_list[0], pod_yaml=pod_yaml,
+                                            failover_node=self.original_control_node)
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+            LOGGER.info("Successfully restored pod by %s way", self.restore_method)
+        if self.restore_node:
+            LOGGER.info("Cleanup: Power on the %s down node.", self.node_name)
+            resp = self.ha_obj.host_power_on(host=self.node_name)
+            assert_utils.assert_true(resp, "Host is not powered on")
+            LOGGER.info("Cleanup: %s is Power on. Sleep for %s sec for pods to join back the"
+                        " node", self.node_name, HA_CFG["common_params"]["pod_joinback_time"])
+            time.sleep(HA_CFG["common_params"]["pod_joinback_time"])
+        # TODO: As cluster restart is not supported until F22A, Need to redeploy cluster after
```

Comment:
so control node restart will not inpact IOs, so redeployment shud not be needed, isn't it?

Reply:
For restart node case it will be needed.